### DISCUSSION TITLE
Editing dynamic plot legend & Li metal anode options

### DISCUSSION THREAD

**Discussion by Jack-Waghorne** [pybamm-team/PyBaMM#1845] [Q&A] [answered]
Hello PyBaMM contributors,

I have created a dynamic plot containing two different power drive cycles over the same period, using the same model. I was hoping to edit the legend to reflect the power cycle used rather than the model which is the default, I was curious if there was a way of doing this?

Along with this, I have been trying to utilise the full or sub-model for lithium metal anodes described in your documentation. With the full model I was confused by what options the model is looking for as I keep getting the error "unexpected keyword argument 'options'". When using the sub-model, is it a simple case of redefining the working electrode and assigning it a new material within that statement?  I was curious if there are any examples of either of these available?

I am new to Python so I apologise if these answers are glaringly obvious. Thank you in advance for your guidance. 

Kind regards
Jack

**Comment by brosaplanella** [comment, marked as answer]
Hi Jack,

You can pass an extra argument `label` to `pybamm.dynamic_plot` to define the labels in the legend:
```python3
pybamm.dynamic_plot(sims, labels=["1", "2", "3", "4"])
```

For the lithium-metal models, you can see an example on how to call them in `examples/scripts/compare_lithium_ion_half_cell.py`.

**Reply by Jack-Waghorne** [reply]
Hello Ferran, 

Thank you very much for your reply. In regards to the example mentioned about lithium anode cells, is the only necessary submodel to convert the model to a lithium anode {"working electrode": "positive"}? Does there not need to be any adjustments to the parameterization to specify this further? 

Best
Jack

**Reply by brosaplanella** [reply]
By setting the option `"working electrode"` PyBaMM sets the relevant submodels for the lithium metal. It also defines the relevant parameter set, but depending on your cell you might want to adjust the parameters. The default parameter set is `Xu2019` so you can call it and see (and modify) the values.

**Reply by Jack-Waghorne** [reply]
I see, I was just overcomplicating it for myself. Thank you for your responses Ferran, much appreciated!

**Reply by brosaplanella** [reply]
No worries!
